vkubelet/env.go

```diff
@@ -347,6 +400,29 @@ loop:
 			continue loop
 		}
 	}
+
+	// enableServiceLinks defaults to true.
+	enableServiceLinks := true
+	if pod.Spec.EnableServiceLinks != nil {
```

Comment:
The kubelet errors out if this is nil, I think we should do the same.

Reply:
My only concern was that some e2e tests might fail, but let's see.